### PR TITLE
Add rudimentary GPG functionality

### PR DESCRIPTION
**Pull request by v-maurer**
Addresses issue #5 
Includes functions to:
- read openpgp.Entity from io.Reader
- decrypt private keys of openpgp.Entity
- sign public key of one Entity with private key of another and export as armored ascii string

### CODE REVIEW

**Comment by v-maurer** [comment]
Two of the functions of issue #5 are still to come, but may need more input on how to interface with the rest of the code.

File: gpg/gpg.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	_, err := buffer.ReadFrom(r)
+	if err != nil {
+		return nil, err
+	}
+
+	gpg := new(GPG)
+	for _, armored := range []bool{false, true} {
+		gpg.serverEntity, err = ReadEntity(bytes.NewReader([]byte(buffer.String())), armored)
+		if err == nil {
+			break
```
Comment (JohannesEbke) [inline gpg/gpg.go:29]: Doesn't this combine with the check below to a return nil, err here?

File: gpg/gpg.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	signedIdentity := ""
+	for _, identity := range clientEntity.Identities {
+		if identity.UserId.Email == signedEMail {
+			signedIdentity = identity.Name
+			break
+		}
+	}
+
+	if signedIdentity == "" {
+		return errors.New(fmt.Sprint("Could find", signedEMail, "in identities of client key"))
```
Comment (JohannesEbke) [inline gpg/gpg.go:60]: Could not find?

File: gpg/helper.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/packet"
+)
+
+// ReadEntity reads a single entity from a reader containing a list of entities.
+func ReadEntity(r io.Reader, armored bool) (*openpgp.Entity, error) {
```
Comment (JohannesEbke) [inline gpg/helper.go:14]: Shouldn't we make these private for now (readEntity?)

**Comment by JohannesEbke** [comment]
Looks good already! Feel free to merge this; even if it's not complete yet.

**Comment by Pharb** [comment]
I will merge the current state of this to master to start working on other issues.

@v-maurer could you please take a look at the comments here, even though this pull request will be already closed?

File: gpg/gpg.go (hunk 1 above)
Comment (v-maurer) [inline gpg/gpg.go:29]: No, I don't think so. It first tries unarmored and then armored, breaking when one option yields no error. But it's probably not the best way to write that part I'll rewrite it a bit clearer.

File: gpg/helper.go (hunk 3 above)
Comment (v-maurer) [inline gpg/helper.go:14]: Yeah sure, if it's not a problem to have dedicated tests for private functions.

Comment (Pharb) [inline gpg/helper.go:14]: @v-maurer testing private functions should be ok according to https://talks.golang.org/2014/testing.slide#20

**Pull request merged**